v2/controllers/pod_watcher.go

```diff
@@ -158,6 +178,7 @@ OUTER2:
 	}
 
 	r.podAddrs[key] = podIPs
+	clientPods.WithLabelValues(r.myNS, r.myName).Set(float64(len(r.podAddrs)))
```

Comment:
It is more efficient to cache `clientPods.WithLabelValues(r.myNS, r.myName)` in `podWatcher` struct.